fpdcc/ccfp-asset-dashboard
# Add authenticated endpoints

## Overview

Closes:
- #105 

### Notes
Found a broken test in the process. I've been submitting PR's against the `asset-ui` branch and CI/CD apparently doesn't run unless it's against the main branch. So there is a small code change in template to fix that.

## Testing Instructions
- Fetch this branch locally and run the tests:
  - `docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app`
- You can also test the review app: https://ccfp-asset-d-authentica-ghhstx.herokuapp.com
- Lastpass credential name: `ccfp-asset-dashboard test user`
- Visit the asset map ui and save some assets. This should work.
- In postman, you can try submitting an unauthenticated request to `localhost:8000/local-assets/` with this request body:
```json
[
    {
        "asset_id": 1,
        "asset_type": "trails"
        "asset_name": "red trail"
        "geom": {
        "type": "LineString",
        "coordinates": [
          [-87.87729052716881, 41.70864290520196],
          [-87.87722544938457, 41.708603704430736],
          [-87.87716885402861, 41.708534150345045],
          [-87.87713091829738, 41.70845537603724]
        ]
      },
    }
]
```
This should give you a 403 error.
